src/Formats/SchemaInferenceUtils.cpp

```diff
@@ -0,0 +1,966 @@
+#include <Formats/SchemaInferenceUtils.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypeDateTime64.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <DataTypes/DataTypeDate.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypeTuple.h>
+#include <DataTypes/DataTypeMap.h>
+#include <DataTypes/DataTypeLowCardinality.h>
+#include <DataTypes/DataTypeNothing.h>
+#include <DataTypes/transformTypesRecursively.h>
+#include <DataTypes/DataTypeObject.h>
+#include <DataTypes/DataTypeFactory.h>
+#include <IO/ReadBufferFromString.h>
+#include <IO/ReadHelpers.h>
+#include <IO/parseDateTimeBestEffort.h>
+#include <IO/PeekableReadBuffer.h>
+
+#include <Core/Block.h>
+#include <Common/assert_cast.h>
+
+namespace DB
+{
+
+static bool checkIfTypesAreEqual(const DataTypes & types)
```

Comment:
In general it's better to use anonymous namespaces for these local functions so you don't write `static` every time.